### PR TITLE
gh-109466: Add ipv6_mapped property to IPv4Address

### LINKED ISSUE #109466: Add ipv6_mapped property to IPv4Address
# Feature or enhancement

### Proposal:

Proposing a `ipv6_mapped` property for IPv4Address. This is the other direction of the `ipv4_mapped` property that already exists in `IPv6Address`. This can be useful if you need an ipv6 representation for an ipv4 address.

### Has this already been discussed elsewhere?

I have already discussed this feature proposal on Discourse

### Links to previous discussion of this feature:

https://discuss.python.org/t/ipv4-to-ipv6-mapped-address/31464

### Linked PRs
* gh-109467

### PR DESCRIPTION
**Pull request by csm10495**
Updated docs and added a corresponding unit test.

* Issue: gh-109466

Fixes #109466.

----
:books: Documentation preview :books:: https://cpython-previews--109467.org.readthedocs.build/

### CODE REVIEW

**Comment by csm10495** [comment]
@asvetlov I think you were a more recent committer here. Any chance for a review? Thanks!

**Comment by csm10495** [comment]
Friendly ping @asvetlov. Thanks.

File: Doc/library/ipaddress.rst [hunk 1]
```diff
@@ -219,6 +219,13 @@ write code that handles both IP versions correctly.  Address objects are
       ``True`` if the address is reserved for link-local usage.  See
       :RFC:`3927`.
 
+   .. attribute:: ipv6_mapped
+
+      :class:`IPv4Address` object representing the IPv4 Mapped IPv6 Address. See :RFC:`4291`.
```
Comment (hugovk) [inline Doc/library/ipaddress.rst]: This is how the RFC writes it:
```suggestion
      :class:`IPv4Address` object representing the IPv4-mapped IPv6 address. See :RFC:`4291`.
```

File: Lib/ipaddress.py [hunk 2]
```diff
@@ -1389,6 +1389,16 @@ def is_link_local(self):
         """
         return self in self._constants._linklocal_network
 
+    @property
+    def ipv6_mapped(self):
+        """Return the IPv6 mapped address.
```
Comment (hugovk) [inline Lib/ipaddress.py]:
```suggestion
        """Return the IPv4-mapped IPv6 address.
```

File: Lib/ipaddress.py [hunk 3]
```diff
@@ -1389,6 +1389,16 @@ def is_link_local(self):
...
+
+        Returns:
+            The IPv4 Mapped IPv6 Address per RFC 4291
```
Comment (hugovk) [inline Lib/ipaddress.py]:
```suggestion
            The IPv4-mapped IPv6 address per RFC 4291.
```

File: Misc/NEWS.d/next/Library/2023-09-15-12-30-21.gh-issue-109466.6ah-aw.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Add the :attr:`ipaddress.IPv4Address.ipv6_mapped` property.
```
Comment (hugovk) [inline Misc/NEWS.d/next/Library/2023-09-15-12-30-21.gh-issue-109466.6ah-aw.rst]:
```suggestion
Add the :attr:`ipaddress.IPv4Address.ipv6_mapped` property, which retuns the IPv4-mapped IPv6 address.
```

If you like, you can append "Patch by [your name]."

File: Doc/whatsnew/3.13.rst [hunk 5]
```diff
@@ -151,6 +151,12 @@ and only logged in :ref:`Python Development Mode <devmode>` or on :ref:`Python
 built on debug mode <debug-build>`.
 (Contributed by Victor Stinner in :gh:`62948`.)
 
+ipaddress
+---------
+
+* Add the :attr:`ipaddress.IPv4Address.ipv6_mapped` property.
```
Comment (hugovk) [inline Doc/whatsnew/3.13.rst]:
```suggestion
* Add the :attr:`ipaddress.IPv4Address.ipv6_mapped` property, which returns the IPv4-mapped IPv6 address.
```

**Review by hugovk — COMMENTED** [review]
I'll let someone else review whether this is needed, but here's some docs suggestions.

**Review by gpshead — APPROVED** [review]
thanks!

**Comment by jstasiak** [comment]
Cross-referencing https://github.com/python/cpython/pull/119814 for visibility.

**Pull request merged**